pkg/cmd/pulumi/pulumi_test.go

```diff
@@ -695,3 +695,234 @@ func TestCheckForUpdate_WorksCorrectlyWithLocalVersions(t *testing.T) {
 	require.Nil(t, stillNilDiag)
 	require.Nil(t, alwaysNilDiag)
 }
+
+//nolint:paralleltest // changes environment variables and globals
+func TestCheckForUpdate_WorksCorrectlyWithVeryOldMajorVersions(t *testing.T) {
+	// Arrange.
```

Comment:
Oh no, it's back :sweat_smile:

Reply:
I still don't like it, but I guess it's fine if it's just this fine.  Just worried that it will then be copied to other files as well by AI/people that are new to the codebase.